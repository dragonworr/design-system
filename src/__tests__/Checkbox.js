import React from 'react'
import renderer from 'react-test-renderer'
import { Checkbox, theme } from '..'
import { shallow, mount, render } from 'enzyme'

const fakeEvent = { target: { checked: true } }
let onClickFn = jest.fn()
const name = 'random_check_box'
const value = 'priceline rules'
const facadeSelector = '[data-test="checkbox-facade"]'
const checkboxSelector = 'input[type="checkbox"]'

describe('Checkbox', () => {
  beforeEach(() => {
    onClickFn.mockReset()
  })

  test('renders without the theme passed specifically', () => {
    const component = <Checkbox name={name} onClick={onClickFn} />
    const wrapper = shallow(component)
    const facade = wrapper.find(facadeSelector)
    const input = wrapper.find(checkboxSelector)
    const json = renderer.create(component).toJSON()

    expect(json.type).toBe('label')
    expect(input.prop('style')).toEqual({
      opacity: 0,
      zIndex: -1,
      position: 'absolute'
    })
    expect(wrapper.children().length).toBe(2)
    expect(facade.children().length).toBe(1)
    expect(json).toMatchSnapshot()
  })

  test('renders with the theme passed specifically', () => {
    const component = <Checkbox name={name} onClick={onClickFn} theme={theme} />
    const wrapper = shallow(component)
    const facade = wrapper.find(facadeSelector)
    const input = wrapper.find(checkboxSelector)
    const json = renderer.create(component).toJSON()

    expect(json.type).toBe('label')
    expect(input.prop('style')).toEqual({
      opacity: 0,
      zIndex: -1,
      position: 'absolute'
    })
    expect(wrapper.children().length).toBe(2)
    expect(facade.children().length).toBe(1)
    expect(json).toMatchSnapshot()
  })

  test('renders checked when defaultChecked prop is passed as true', () => {
    const component = (
      <Checkbox name={name} defaultChecked={true} onClick={onClickFn} />
    )
    const wrapper = shallow(component)
    const input = wrapper.find(checkboxSelector)
    const json = renderer.create(component).toJSON()

    expect(input.prop('defaultChecked')).toBe(true)
    expect(json).toMatchSnapshot()
  })

  test('renders disabled when disabled props is passed as true', () => {
    const component = (
      <Checkbox name={name} disabled={true} onClick={onClickFn} />
    )
    const componentChecked = (
      <Checkbox
        name={name}
        disabled={true}
        defaultChecked={true}
        onClick={onClickFn}
      />
    )

    const json = renderer.create(component).toJSON()
    const jsonChecked = renderer.create(componentChecked).toJSON()

    expect(json).toMatchSnapshot()
    expect(jsonChecked).toMatchSnapshot()
  })

  /** Not Testing wrapper.simulate('click) because enzyme won't pass it through to the checkbox
   * We have to test the actual element receiving the event and the label 'click' causes a 'change'
   * for the nested input. This also means we cannot accurately test that disabled checkboxes ignore clicks.
   */
  test('updates and fires callback when clicked (when enabled)', () => {
    const component = <Checkbox name={name} onClick={onClickFn} />
    const wrapper = shallow(component)
    const json = renderer.create(component).toJSON()

    wrapper.find('input').simulate('change', fakeEvent)
    expect(onClickFn).toHaveBeenCalled()
    expect(json).toMatchSnapshot()
  })
})
