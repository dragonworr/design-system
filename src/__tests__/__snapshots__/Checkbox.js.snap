// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders checked when defaultChecked prop is passed as true 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #fff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  font-size: 10px;
  color: #687B8E;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0 input:not(:checked) + [data-test='checkbox-facade'] {
  background-color: #fff;
}

.c0 input:not(:checked) + [data-test='checkbox-facade']:hover {
  border-color: #007aff;
}

.c0 input:checked + [data-test='checkbox-facade'] {
  background-color: #007aff;
  border-color: #007aff;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  cursor: pointer;
  border: 2px solid #d1d6db;
}

<label
  className="c0"
  color="gray"
  data-test="design-system-checkbox"
  disabled={false}
  fontSize="10px"
  fontWeight="bold"
  htmlFor="random_check_box"
>
  <input
    defaultChecked={true}
    disabled={false}
    id="random_check_box"
    name="random_check_box"
    onChange={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
    type="checkbox"
  />
  <div
    className="c1"
    data-test="checkbox-facade"
    disabled={false}
    size={20}
  >
    <svg
      className="c2"
      color="#fff"
      fill="currentcolor"
      height={16}
      viewBox="0 0 20 15"
      width={16}
    >
      <path
        d="M9.26 12.53l10.188-9.55a1.657 1.657 0 0 0 0-2.469c-.736-.681-1.93-.681-2.665 0l-10.188 9.55-3.378-3.13c-.735-.682-1.93-.682-2.665 0a1.657 1.657 0 0 0 0 2.469l3.379 3.13L6.595 15l2.665-2.47z"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox renders disabled when disabled props is passed as true 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #fff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  font-size: 10px;
  color: #687B8E;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 0.7;
}

.c0 input:not(:checked) + [data-test='checkbox-facade'] {
  background-color: #fff;
  opacity: 0.7;
}

.c0 input:checked + [data-test='checkbox-facade'] {
  background-color: #d1d6db;
  border-color: #d1d6db;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  cursor: pointer;
  border: 2px solid #d1d6db;
}

<label
  className="c0"
  color="gray"
  data-test="design-system-checkbox"
  disabled={true}
  fontSize="10px"
  fontWeight="bold"
  htmlFor="random_check_box"
>
  <input
    defaultChecked={undefined}
    disabled={true}
    id="random_check_box"
    name="random_check_box"
    onChange={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
    type="checkbox"
  />
  <div
    className="c1"
    data-test="checkbox-facade"
    disabled={true}
    size={20}
  >
    <svg
      className="c2"
      color="#fff"
      fill="currentcolor"
      height={16}
      viewBox="0 0 20 15"
      width={16}
    >
      <path
        d="M9.26 12.53l10.188-9.55a1.657 1.657 0 0 0 0-2.469c-.736-.681-1.93-.681-2.665 0l-10.188 9.55-3.378-3.13c-.735-.682-1.93-.682-2.665 0a1.657 1.657 0 0 0 0 2.469l3.379 3.13L6.595 15l2.665-2.47z"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox renders disabled when disabled props is passed as true 2`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #fff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  font-size: 10px;
  color: #687B8E;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 0.7;
}

.c0 input:not(:checked) + [data-test='checkbox-facade'] {
  background-color: #fff;
  opacity: 0.7;
}

.c0 input:checked + [data-test='checkbox-facade'] {
  background-color: #d1d6db;
  border-color: #d1d6db;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  cursor: pointer;
  border: 2px solid #d1d6db;
}

<label
  className="c0"
  color="gray"
  data-test="design-system-checkbox"
  disabled={true}
  fontSize="10px"
  fontWeight="bold"
  htmlFor="random_check_box"
>
  <input
    defaultChecked={true}
    disabled={true}
    id="random_check_box"
    name="random_check_box"
    onChange={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
    type="checkbox"
  />
  <div
    className="c1"
    data-test="checkbox-facade"
    disabled={true}
    size={20}
  >
    <svg
      className="c2"
      color="#fff"
      fill="currentcolor"
      height={16}
      viewBox="0 0 20 15"
      width={16}
    >
      <path
        d="M9.26 12.53l10.188-9.55a1.657 1.657 0 0 0 0-2.469c-.736-.681-1.93-.681-2.665 0l-10.188 9.55-3.378-3.13c-.735-.682-1.93-.682-2.665 0a1.657 1.657 0 0 0 0 2.469l3.379 3.13L6.595 15l2.665-2.47z"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox renders with the theme passed specifically 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #fff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  font-size: 10px;
  color: #687B8E;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0 input:not(:checked) + [data-test='checkbox-facade'] {
  background-color: #fff;
}

.c0 input:not(:checked) + [data-test='checkbox-facade']:hover {
  border-color: #007aff;
}

.c0 input:checked + [data-test='checkbox-facade'] {
  background-color: #007aff;
  border-color: #007aff;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  cursor: pointer;
  border: 2px solid #d1d6db;
}

<label
  className="c0"
  color="gray"
  data-test="design-system-checkbox"
  disabled={false}
  fontSize="10px"
  fontWeight="bold"
  htmlFor="random_check_box"
>
  <input
    defaultChecked={undefined}
    disabled={false}
    id="random_check_box"
    name="random_check_box"
    onChange={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
    type="checkbox"
  />
  <div
    className="c1"
    data-test="checkbox-facade"
    disabled={false}
    size={20}
  >
    <svg
      className="c2"
      color="#fff"
      fill="currentcolor"
      height={16}
      viewBox="0 0 20 15"
      width={16}
    >
      <path
        d="M9.26 12.53l10.188-9.55a1.657 1.657 0 0 0 0-2.469c-.736-.681-1.93-.681-2.665 0l-10.188 9.55-3.378-3.13c-.735-.682-1.93-.682-2.665 0a1.657 1.657 0 0 0 0 2.469l3.379 3.13L6.595 15l2.665-2.47z"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox renders without the theme passed specifically 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #fff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  font-size: 10px;
  color: #687B8E;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0 input:not(:checked) + [data-test='checkbox-facade'] {
  background-color: #fff;
}

.c0 input:not(:checked) + [data-test='checkbox-facade']:hover {
  border-color: #007aff;
}

.c0 input:checked + [data-test='checkbox-facade'] {
  background-color: #007aff;
  border-color: #007aff;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  cursor: pointer;
  border: 2px solid #d1d6db;
}

<label
  className="c0"
  color="gray"
  data-test="design-system-checkbox"
  disabled={false}
  fontSize="10px"
  fontWeight="bold"
  htmlFor="random_check_box"
>
  <input
    defaultChecked={undefined}
    disabled={false}
    id="random_check_box"
    name="random_check_box"
    onChange={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
    type="checkbox"
  />
  <div
    className="c1"
    data-test="checkbox-facade"
    disabled={false}
    size={20}
  >
    <svg
      className="c2"
      color="#fff"
      fill="currentcolor"
      height={16}
      viewBox="0 0 20 15"
      width={16}
    >
      <path
        d="M9.26 12.53l10.188-9.55a1.657 1.657 0 0 0 0-2.469c-.736-.681-1.93-.681-2.665 0l-10.188 9.55-3.378-3.13c-.735-.682-1.93-.682-2.665 0a1.657 1.657 0 0 0 0 2.469l3.379 3.13L6.595 15l2.665-2.47z"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox updates and fires callback when clicked (when enabled) 1`] = `
.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: #fff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  font-size: 10px;
  color: #687B8E;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0 input:not(:checked) + [data-test='checkbox-facade'] {
  background-color: #fff;
}

.c0 input:not(:checked) + [data-test='checkbox-facade']:hover {
  border-color: #007aff;
}

.c0 input:checked + [data-test='checkbox-facade'] {
  background-color: #007aff;
  border-color: #007aff;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  cursor: pointer;
  border: 2px solid #d1d6db;
}

<label
  className="c0"
  color="gray"
  data-test="design-system-checkbox"
  disabled={false}
  fontSize="10px"
  fontWeight="bold"
  htmlFor="random_check_box"
>
  <input
    defaultChecked={undefined}
    disabled={false}
    id="random_check_box"
    name="random_check_box"
    onChange={[Function]}
    style={
      Object {
        "opacity": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
    type="checkbox"
  />
  <div
    className="c1"
    data-test="checkbox-facade"
    disabled={false}
    size={20}
  >
    <svg
      className="c2"
      color="#fff"
      fill="currentcolor"
      height={16}
      viewBox="0 0 20 15"
      width={16}
    >
      <path
        d="M9.26 12.53l10.188-9.55a1.657 1.657 0 0 0 0-2.469c-.736-.681-1.93-.681-2.665 0l-10.188 9.55-3.378-3.13c-.735-.682-1.93-.682-2.665 0a1.657 1.657 0 0 0 0 2.469l3.379 3.13L6.595 15l2.665-2.47z"
      />
    </svg>
  </div>
</label>
`;
